client/lib/domains/is-subdomain.ts

```diff
@@ -0,0 +1,16 @@
+import { getRootDomain } from 'calypso/lib/domains/utils';
+
+export function isSubdomain( domainName: string ) {
+	if ( ! domainName || domainName === '' ) {
+		return false;
+	}
+	domainName = domainName.startsWith( 'www.' ) ? domainName.slice( 4 ) : domainName;
+	const isValidSubdomain = Boolean(
+		domainName &&
+			domainName.match(
+				/^([a-z0-9_]([a-z0-9\-_]*[a-z0-9_])?\.)+[a-z0-9]([a-z0-9-]*[a-z0-9])?\.[a-z]{2,63}$/
+			)
```

Comment:
```suggestion
		domainName.match(
			/^([a-z0-9_]([a-z0-9\-_]*[a-z0-9_])?\.)+[a-z0-9]([a-z0-9-]*[a-z0-9])?\.[a-z]{2,63}$/
		)
```

We already check if `domainName` is null or empty above, so I guess this check here is redundant?